### ISSUE TITLE
generateSarif | INPUT_GENERATESARIF causes crash

### ISSUE THREAD

**Issue by mschwager** [returntocorp/sgrep-action#346]
Background: https://r2c-community.slack.com/archives/CK86BJ5DW/p1628266506046400

When running with SARIF output, `semgrep_agent` crashes under certain repositories.

To repro:

- Visit https://github.com/fleetdm/fleet/tree/93a10e6f943f260e03a73d079b2f6074f659f8a5
- Run `git clone https://github.com/fleetdm/fleet.git`
- `cd fleet` and run `semgrep_agent`
- First, notice the following doesn't crash: `python -m semgrep_agent --config p/kubernetes`
- Second, notice the following does crash: `INPUT_GENERATESARIF=1 python -m semgrep_agent --config p/kubernetes`

```
=== looking for current issues in 1857 files
| Semgrep exited with 1 error:
|    charts/fleet/templates/deployment.yaml:8 Could not parse deployment.yaml as yaml

semgrep-core message:

	Other parsing error: Expected a valid scalar, got
| 6 current issues found
| No ignored issues found
=== not looking at pre-existing issues since all files with current issues are newly created
=== re-running scan to generate a SARIF report
An unexpected error occurred
ERROR:root:Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "~/r2c/semgrep-action/src/semgrep_agent/main.py", line 176, in main
    protected_main(**locals())
  File "~/r2c/semgrep-action/src/semgrep_agent/main.py", line 355, in protected_main
    timeout=(timeout if timeout > 0 else None),
  File "~/r2c/semgrep-action/src/semgrep_agent/semgrep.py", line 356, in scan
    timeout=timeout,
  File "~/r2c/semgrep-action/src/semgrep_agent/semgrep.py", line 266, in get_findings
    rewrite_sarif_file(sarif_path)
  File "~/r2c/semgrep-action/src/semgrep_agent/semgrep.py", line 98, in rewrite_sarif_file
    sarif_results = json.load(sarif_file)
  File "~/.pyenv/versions/3.7.0/lib/python3.7/json/__init__.py", line 296, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "~/.pyenv/versions/3.7.0/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "~/.pyenv/versions/3.7.0/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "~/.pyenv/versions/3.7.0/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This was originally noted here: https://github.com/fleetdm/fleet/runs/3258407491

My first suspicion would be something to do with this comment: `# FIXME: This will crash when running on thousands of files due to command length limit` and this log line: `found 2037 files in the paths to be scanned`.

**Comment by pyth0n1c** [comment]
I am seeing the same issue on a different repo.  I can also confirm that 

```
- uses: returntocorp/semgrep-action@v1
        with:
          generateSarif: "1"
          config: >- # more at semgrep.dev/explore
            p/security-audit
            p/secrets

``` 
fails with the same error noted above while:

```
- uses: returntocorp/semgrep-action@v1
        with:
          config: >- # more at semgrep.dev/explore
            p/security-audit
            p/secrets
 
```
works as expected.

**Comment by brendongo** [comment]
Fixed by #351
